net/ip.go

```diff
@@ -309,3 +311,32 @@ func (ip *IP) Next() *IP {
 
 	return newIP
 }
+
+// MaskLastNBits masks the last n bits of an IP address
+func (ip *IP) MaskLastNBits(n uint8) *IP {
+	ip = ip.copy()
+
+	if ip.isLegacy {
+		ip.maskLastNBitsIPv4(n)
+		return ip
+	}
+
+	ip.maskLastNBitsIPv6(n)
+	return ip
+}
+
+func (ip *IP) maskLastNBitsIPv4(n uint8) {
```

Comment:
I'd add a check if n <= 32 here

Reply:
For what? Cutting off more than the address length returns 0.0.0.0 and is what I would expect. No?